### PR TITLE
Denis jan19_2 homework

### PR DESCRIPTION
**Pull request by frolovde**

### CODE REVIEW

File: tests/Orange_HRM_Add_Employee.py [hunk 1]
```diff
@@ -35,28 +35,45 @@ def test_emp_add_user(self):
...
+
         wait.until(EC.presence_of_element_located([By.CSS_SELECTOR, '#empsearch_employee_name_empName.inputFormatHint']))
         driver.find_element_by_id("btnAdd").click()
         # TODO Need to figure out why the following Waits are not working: #wait.until(EC.url_contains('/viewEmployeeList'))
         #  #wait.until(EC.element_to_be_clickable([By.ID, "btnAdd"]))
+
+        # Two more Waits for practice
+
         wait.until(EC.url_to_be(desired_url))
-        wait.until(EC.presence_of_element_located([By.ID, "middleName"]))
```
Comment (frolovde) [inline tests/Orange_HRM_Add_Employee.py]: I updated this to First name

File: tests/Orange_HRM_Add_Employee.py [hunk 2]
```diff
@@ -35,28 +35,45 @@ def test_emp_add_user(self):
...
+        wait.until(EC.presence_of_element_located([By.ID, "firstName"]))
         driver.find_element_by_id("firstName").send_keys("Denis")
         driver.find_element_by_id("lastName").send_keys("Frolov")
         employee_id = driver.find_element_by_id("employeeId").get_attribute("value")
-        print(employee_id)
```
Comment (frolovde) [inline tests/Orange_HRM_Add_Employee.py]: Removed print, I was making sure it was capturing it.

File: tests/Orange_HRM_Add_Employee.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        driver.find_element_by_id("user_name").send_keys("Denisfrolov" + employee_id)
+        driver.find_element_by_id("user_password").send_keys("password")
+        driver.find_element_by_id("re_password").send_keys("password")
+        driver.find_element_by_id("status").click()
+        driver.find_element_by_id("btnSave").click()
+
+        # 2 Waits for practice
+
+        wait.until(EC.url_changes)
+        wait.until((EC.presence_of_element_located([By.ID, "btnSave"])))
```
Comment (EllieSky) [inline tests/Orange_HRM_Add_Employee.py:72]: Actually this same locator is present on both pages, so this wait wouldn’t be helpful

**Review by EllieSky — COMMENTED** [review]
Good use of Explicit wait.
Improve your code further by using variables instead of hardcoded values. For example store your first name and last name into variables, and reuse those throughout your script.

File: tests/Orange_HRM_Add_Employee.py (hunk 3 above)
Comment (frolovde) [reply tests/Orange_HRM_Add_Employee.py:72]: Ahh, didn't notice that, thanks. Changed it to this:

        # Making sure Personal Details banner is displayed
        wait.until((EC.presence_of_element_located([By.XPATH, "//h1[contains(text(),'Personal Details')]"])))

File: tests/Orange_HRM_Add_Employee.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        wait.until(EC.visibility_of_element_located([By.XPATH, "//a[contains(text(),'Logout')]"]))
+        driver.find_element_by_xpath("//a[contains(text(),'Logout')]").click()
+        self.assertTrue(driver.current_url, expected_login_url)
+
+# Logging in with a newly created user
+
+        self.login(user_name + employee_id, new_password)
+        wait.until(EC.presence_of_element_located([By.ID, "welcome"]))
+        welcome = driver.find_element_by_id("welcome").text
+        self.assertEqual("Welcome Denis", welcome)
```
Comment (EllieSky) [inline tests/Orange_HRM_Add_Employee.py:94]: “Welcome “ + first_name

File: tests/Orange_HRM_Add_Employee.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    def test_emp_add_user(self):
+        self.login('admin', 'password')
+
+        # Test Variables
+        driver = self.driver
+        desired_url = 'http://hrm-online.portnov.com/symfony/web/index.php/pim/addEmployee'
+        wait = WebDriverWait(driver, 15)
+        first_name = "Denis"
+        last_name = "Frolov"
+        user_name = "Denisfrolov"
```
Comment (EllieSky) [inline tests/Orange_HRM_Add_Employee.py:42]: user_name = (first_name + last_name).lower()

**Review by EllieSky — COMMENTED** [review]
Almost perfect

**Pull request closed without merging**